### PR TITLE
avoid calling should_compact_on_launch_function

### PR DESCRIPTION
**Pull request by jpsim**
when a Realm is already open at that path. Fixes Cocoa #5010 (https://github.com/realm/realm-cocoa/issues/5010).

### CODE REVIEW

**Comment by bdash** [comment]
Please don't use "Fixes org/repo#xyz" across repositories. It'd result in the original issue being closed as soon as the PR is merged into this repository, while further work is still required for the fix to be made available to Cocoa.

**Comment by jpsim** [comment]
Alright. I reworded the comment to still convey what I meant without making GitHub think it actually fixes it.

**Comment by jpsim** [comment]
@tgoyne think you could review this one?

File: src/shared_realm.cpp [hunk 1]
```diff
@@ -161,7 +161,9 @@ void Realm::open_with_config(const Config& config,
             };
             shared_group = std::make_unique<SharedGroup>(*history, options);
 
-            if (realm && config.should_compact_on_launch_function) {
+            auto coordinator = RealmCoordinator::get_existing_coordinator(config.path);
+
+            if (realm && config.should_compact_on_launch_function && !coordinator->has_initialized_realms()) {
```
Comment (tgoyne) [inline src/shared_realm.cpp]: I think this block makes more sense in Realm's constructor than here. We don't want to do the compact when we're calling `open_with_config()` from anywhere but the constructor (it's used for things like opening the helper SGs for the notification worker), and there we already have a pointer to the coordinator.

**Comment by jpsim** [comment]
Thanks for the tip, that's much better,

**Comment by bdash** [comment]
@jpsim, not sure if you noticed but there are a number of test failures on CI that need to be addressed before this can be merged.

**Comment by jpsim** [comment]
CI is now passing, but I'm still writing a regression test, porting it over from the cocoa PR.

**Comment by jpsim** [comment]
I've added a regression test, so this is now ready for review again.

**Pull request merged**